com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/DebugUtility.java

```diff
@@ -313,6 +315,26 @@ public static void resumeThread(ThreadReference thread) {
     }
 
     /**
+     * Compute the stack frame model from the ThreadReference.
+     *
+     * @param thread
+     *            the thread reference
+     * @return the list of the computed stack frame model
+     */
+    public static List<IStackFrame> computeStackFrame(ThreadReference thread) {
+        List<IStackFrame> result = new ArrayList<IStackFrame>();
+        try {
+            List<StackFrame> rawFrames = thread.frames();
+            for (StackFrame frame : rawFrames) {
```

Comment:
list all  stack frame is very time-consuming, currently stack frame is listed using the pagination way.